waspc/data/Generator/templates/react-app/src/actions/index.js

```diff
@@ -0,0 +1,74 @@
+import {
+  useMutation,
+  useQueryClient,
+} from 'react-query'
+
+export { configureQueryClient } from '../queryClient'
+
+export function useAction(actionFn, actionOptions) {
+  const queryClient = useQueryClient();
+
+  // TODO(filip): Do we allow all react query options or should we consider a list of allowed options?
+  let options = {}
+
+  if (actionOptions?.optimisticUpdates) {
+    const optimisticUpdatesConfig = makeOptimisticUpdatesConfig(actionOptions.optimisticUpdates)
+    options = makeOptimisticUpdateOptions(queryClient, optimisticUpdatesConfig)
+  }
+
+  return useMutation(actionFn, options)
+}
+
+
+function makeOptimisticUpdatesConfig(optimisticUpdatesConfig) {
+  return optimisticUpdatesConfig.map(({ getQuery, ...rest }) => ({
+    getQuery: (item) => parseQueryKey(getQuery(item)),
+    ...rest,
+  }))
+}
```

Comment:
So you basically take already existing config and modify it so that getQuery works a bit differently. I am guessing that is so because we want to have a bit different interface compared to what react-query does? So what this function is really doing is transforming our optimisticUpdatesConfig into react query's optimisticUpdatesConfig?
If so, we could name it that way:
```js
makeRqOptimisticUpdatesConfig(waspOptimisticUpdatesConfig) {
  ...
}
```

Reply:
This function could also maybe benefit from a bit of comments to explain what it takes and what it returns, since we don't have any types.

Reply:
I agree with both comments (will rename the function and add explanations).